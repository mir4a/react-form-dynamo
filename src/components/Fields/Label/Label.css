.Label {
  margin-bottom: 10px;
}

.Label-text {
  font-size: 14px;
  font-weight: 700;
}

.Label__block {
  display: block;
}