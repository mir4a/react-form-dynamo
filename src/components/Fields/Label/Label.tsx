import React, { memo, ReactNode } from 'react';
import classNames from 'classnames';
import './Label.css';

interface Props {
  label: string;
  forId?: string;
  children?: ReactNode;
  className?: string;
}
export default memo(function Label(props: Props) {
  const { label, forId, children, className } = props;
  const labelClassNames = classNames('Label', className);

  return (
    <label htmlFor={forId} className={labelClassNames}>
      {children}
      <span className="Label-text">{label}</span>
    </label>
  );
});
