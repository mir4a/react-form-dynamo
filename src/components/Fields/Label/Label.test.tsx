import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';

it('renders correctly', () => {
  const tree = renderer.create(<Label label="Radio Buttons" />).toJSON();
  expect(tree).toMatchSnapshot();
});
