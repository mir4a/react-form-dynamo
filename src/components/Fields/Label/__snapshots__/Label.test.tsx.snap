// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="Label"
>
  <span
    className="Label-text"
  >
    Radio Buttons
  </span>
</label>
`;
